packages/api/test/__fixtures__/sdk/petstore/index.ts

```diff
@@ -0,0 +1,617 @@
+    return this.core.fetch('/user/createWithArray', 'post', body);
+  }
+
+  /**
+   * Creates list of users with given input array
+   *
+   */
+  createUsersWithListInput<T = unknown>(body: CreateUsersWithListInputBodyParam): Promise<T> {
+    return this.core.fetch('/user/createWithList', 'post', body);
+  }
+
+  /**
+   * This can only be done by the logged in user.
+   *
+   * @summary Updated user
+   */
+  updateUser<T = unknown>(body: User, metadata: UpdateUserMetadataParam): Promise<T> {
+    return this.core.fetch('/user/{username}', 'put', body, metadata);
+  }
+
+  /**
+   * This can only be done by the logged in user.
+   *
+   * @summary Delete user
+   */
+  deleteUser<T = unknown>(metadata: DeleteUserMetadataParam): Promise<T> {
+    return this.core.fetch('/user/{username}', 'delete', metadata);
+  }
+
+  /**
+   * Multiple status values can be provided with comma separated strings
+   *
+   * @summary Finds Pets by status
+   */
+  findPetsByStatus(metadata: FindPetsByStatusMetadataParam): Promise<FindPetsByStatus_Response_200> {
+    return this.core.fetch('/pet/findByStatus', 'get', metadata);
+  }
+
+  /**
+   * Muliple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
+   *
+   * @summary Finds Pets by tags
+   */
+  findPetsByTags(metadata: FindPetsByTagsMetadataParam): Promise<FindPetsByTags_Response_200> {
+    return this.core.fetch('/pet/findByTags', 'get', metadata);
+  }
+
+  /**
+   * Returns a single pet
+   *
+   * @summary Find pet by ID
+   */
+  getPetById(metadata: GetPetByIdMetadataParam): Promise<Pet> {
+    return this.core.fetch('/pet/{petId}', 'get', metadata);
+  }
+
+  /**
+   * Uploads an image
+   *
+   */
+  uploadFile(body: UploadFileBodyParam, metadata: UploadFileMetadataParam): Promise<ApiResponse>;
+  /**
+   * Uploads an image
+   *
+   */
+  uploadFile(metadata: UploadFileMetadataParam): Promise<ApiResponse>;
+  /**
+   * Uploads an image
+   *
+   */
+  uploadFile(body?: UploadFileBodyParam, metadata?: UploadFileMetadataParam): Promise<ApiResponse> {
+    return this.core.fetch('/pet/{petId}/uploadImage', 'post', body, metadata);
+  }
+
+  /**
+   * Place an order for a pet
+   *
+   */
+  placeOrder(body: Order): Promise<Order> {
+    return this.core.fetch('/store/order', 'post', body);
+  }
+
+  /**
+   * For valid response try integer IDs with value >= 1 and <= 10. Other values will generated exceptions
+   *
+   * @summary Find purchase order by ID
+   */
+  getOrderById(metadata: GetOrderByIdMetadataParam): Promise<Order> {
+    return this.core.fetch('/store/order/{orderId}', 'get', metadata);
+  }
+
+  /**
+   * Logs user into the system
+   *
+   */
+  loginUser(metadata: LoginUserMetadataParam): Promise<LoginUser_Response_200> {
+    return this.core.fetch('/user/login', 'get', metadata);
+  }
+
+  /**
+   * Get user by user name
+   *
+   */
+  getUserByName(metadata: GetUserByNameMetadataParam): Promise<User> {
```

Comment:
all these named methods are so nice and clean. I hope it will be one more DX reason to write better oas files.